Source/WebKit/UIProcess/WebBackForwardList.cpp

```diff
@@ -270,6 +270,17 @@ void WebBackForwardList::clearProvisionalItem(WebBackForwardListFrameItem& frame
     m_provisionalIndex = std::nullopt;
 }
 
+void WebBackForwardList::commitProvisionalItem(WebBackForwardListFrameItem& frameItem)
+{
+    if (!m_provisionalIndex)
+        return;
+
+    if (m_entries[*m_provisionalIndex].ptr() != frameItem.backForwardListItem())
```

Comment:
May as well add a bounds check here and assert not reached and early return if it's out of bounds.